app/src/main/res/layout/tool_details_fragment.xml

```diff
@@ -141,12 +143,23 @@
                     app:layout_constraintTop_toBottomOf="@id/action_add"
                     app:visibleIf="@{tool.isAdded()}" />
 
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/action_tool_training"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginLeft="@dimen/tool_details_margin_horizontal"
+                    android:layout_marginRight="@dimen/tool_details_margin_horizontal"
+                    android:onClick="@{() -> fragment.openTraining(manifest)}"
+                    android:text="@string/action_tools_open_training"
+                    app:goneIf="@{!hasTraining}"
```

Comment:
now that the manifest is present in data binding, this can be changed to `app:visibleIf="@{!manifest.tips.isEmpty &amp;&amp; BuildConfig.DEBUG}"`

Reply:
up to you whether you use `app:visibleIf` or `app:goneIf`